packages/astro/src/astro-components/control/Protect.astro

```diff
@@ -0,0 +1,38 @@
+---
+import {
+  CheckAuthorizationWithCustomPermissions,
+  OrganizationCustomPermissionKey,
+  OrganizationCustomRoleKey,
+} from "@clerk/types";
+
+type Props =
+  | {
+      condition?: never;
+      role: OrganizationCustomRoleKey;
+      permission?: never;
+    }
+  | {
+      condition?: never;
+      role?: never;
+      permission: OrganizationCustomPermissionKey;
+    }
+  | {
+      condition: (has: CheckAuthorizationWithCustomPermissions) => boolean;
+      role?: never;
+      permission?: never;
+    }
+  | {
+      condition?: never;
+      role?: never;
+      permission?: never;
+    };
+
+const { has, userId } = Astro.locals.auth();
+const isUnauthorized =
+  !userId ||
+  (typeof Astro.props.condition === "function" &&
+    !Astro.props.condition(has)) ||
+  ((Astro.props.role || Astro.props.permission) && !has(Astro.props));
```

Comment:
nit: What do you think about reversing the logic here? `isAuthorized` feels clearer than `isUnauthorized` (default to unauthorized in the logic).

Reply:
Agree, that this is hard to grok

Reply:
That's a fair point, I will added it as a todo if that's ok